.github/workflows/test.yml

```diff
@@ -371,6 +427,13 @@ jobs:
         with:
           sarif_file: ./hadolint.sarif
       - uses: ./.github/actions/setup-zeebe
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-docker-checks-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-docker-checks-
```

Comment:
What's the difference between this and setting up `cache: maven` in the `setup-zeebe`'s `setup-java` action?

Reply:
this way we have specific caches per job, as not all jobs build all modules, this reduces the overhead per job but increased the total space used for caches, as we have more caches in total. Not sure yet what the better tradeoff is. 🤔

Reply:
I see - I just figured this was the same computation the setup-java action did :smile: 

:wrench: We could still push this into the `setup-zeebe` action

At any rate, I'm fine with trying it out again - do we have some idea how to deal with broken caches this time though? Or do we hope it just won't happen as often?

Reply:
I think we can use https://github.com/actions/gh-actions-cache#readme for that

Reply:
I would expect a build that has a broken state of the m2 repo to fail, thus that state of the broken m2 repo shouldn't get cached going forward. If we hit them, there is also an API to delete them :D https://docs.github.com/en/rest/actions/cache#delete-a-github-actions-cache-for-a-repository-using-a-cache-id
so we could setup a post action to delete broken caches if we see that becoming an issue

Reply:
readme added f1cc42c48da3624e152094cf4aef4cdc67be6d69